File: Data Structures & Algorithms/sliding-window-maximum/submission-2.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        dq = deque()
        ans = []
        i = 0
        for j in range(len(nums)):
            while(dq and dq[-1][0]<nums[j]):
                dq.pop()
            dq.append((nums[j], j))
            if(dq and dq[0][1]<i):
                dq.popleft()
            if(j-i+1==k):
                ans.append(dq[0][0])
                i+=1
        return ans

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int>bucket;
        priority_queue<pair<int, int>>max_pq;
        vector<int> ans;
        for(auto it: nums) bucket[it]++;
        for(auto it: bucket) max_pq.push({it.second, it.first});
        for (int i=0;i<k;i++) {
            ans.push_back(max_pq.top().second);
            max_pq.pop();
        }
        return ans;
    }
};
